core.md

```diff
@@ -130,83 +179,79 @@ default can save a significant amount of memory, particularly if you
+opacity below for details.
 
-### Array type definitions
+### Struct Arrays
 
-You can create a type definition for an array using the `arrayType`
-method that is defined on every other type definition object.  So for
-example an array of 32 points could be created like so:
+For each struct type, a dynamic array of instances of that type can be
+created using the type's `.array` constructor:
 
 ```js
-var PointArrayType = PointType.arrayType(32);
+const Point = new StructType({x: float64, y: float64});
+let points = new Point.array();
 ```
 
-Arrays can be multidimensional, so for example you might define a type
-for a 1024x768 image like so:
+The `array` method supports the same overloads as the `Array` constructor:
+invoking it without arguments creates an empty array, invoking it with a single numeric argument creates an array with a length
+determined by that argument, every other combination of arguments uses the
+arguments to initialize elements in the instance:
 
 ```js
-var ColorType = new StructType({r: uint8, g: uint8,
-                                b: uint8, a: uint8});
-var ImageType = ColorType.arrayType(1024).arrayType(768);
+new Point.array(3); // [Point, Point, Point]
+new Point.array(p1, p2, {x: 10, y: 20}); // [Point, Point, Point]
 ```
 
-*NOTE:* Issue nikomatsakis/typed-objects-explainer#1 proposes to
-change some details of this section.
-
 ## Typed objects: instantiating struct types
 
-You can create an instance of a struct or array type using the `new`
+You can create an instance of a struct type using the `new`
 operator:
 
 ```js
-var line = new LineType();
+const PointType = new StructType({x: float64, y: float64});
+const LineType = new StructType({from: PointType, to: PointType});
+let line = new LineType();
 console.log(line.from.x); // logs 0
 ```
 
 The resulting object is called a *typed object*: it will have the
-fields specified in `LineType`. Each field will be initialized to an
+fields specified in `Line`. Each field will be initialized to an
 appropriate default value based on its type (e.g., numbers are
 initialized to 0, fields of type `any` are initialized to `undefined`,
 and so on). Fields with structural type (like `from` and `to` in this
 case) are recursively initialized.
 
-When creating a new typed object, you can also supply an "example
+When creating a new typed object, you can also supply a "source
 object". This object will be used to extract the initial values for
 each field:
 
 ```js
-var line1 = new LineType({from: {x: 1, y: 2},
+let line1 = new LineType({from: {x: 1, y: 2},
                           to: {x: 3, y: 4}});
 console.log(line1.from.x); // logs 1
 
-var line2 = new LineType(line1);
+let line2 = new LineType(line1);
 console.log(line2.from.x); // also logs 1
 
-var PointsType = PointType.array(2);
-var array = new PointsType([{x: 1, y: 2}, {x: 3, y: 4}]);
+let array = new PointType.array([{x: 1, y: 2}, {x: 3, y: 4}]);
```

Comment:
Is this right? It seems inconsistent with what you wrote about array constructor above. I'm not sure this example belongs here anymore